src/pip/_internal/index.py

```diff
@@ -258,20 +258,119 @@ def _get_html_page(link, session=None):
 
 class CandidateEvaluator(object):
 
+    """
+    Responsible for filtering and sorting candidates for installation based
+    on what tags are valid.
+    """
+
     def __init__(
         self,
         valid_tags,          # type: List[Pep425Tag]
         prefer_binary=False  # type: bool
-
     ):
         # type: (...) -> None
         self._prefer_binary = prefer_binary
         self._valid_tags = valid_tags
 
-    def is_wheel_supported(self, wheel):
+        self._py_version_re = re.compile(r'-py([123]\.?[0-9]?)$')
```

Comment:
Why not use a class variable (or an out-of-class global)? I don’t think this needs to be instantiated separately for each instance.

Reply:
Thanks a lot for taking a look at this PR.

Re: this line, I was being sensitive to this issue (https://github.com/pypa/pip/pull/6346), combined with the fact that `PackageFinder` is generally only instantiated once per pip invocation (in `RequirementCommand._build_package_finder()`). If it were multiple times, I wouldn't hesitate. If you still don't think it should be done this way, I can make it class-level again.

Reply:
I see. It might be worthwhile to add a comment noting that the choice in intentional (and that it’s fine since the class is only instantiated once).